File: RGBEngine/src/main/java/extras/behaviors/QuitOnEscapeBehavior.java
package extras.behaviors;

import engine.core.Behavior.Entity;
import engine.core.Game;
import engine.core.Input;
import static org.lwjgl.glfw.GLFW.GLFW_KEY_ESCAPE;

public class QuitOnEscapeBehavior extends Entity {

    static {
        Game.declareSystem(QuitOnEscapeBehavior.class, QuitOnEscapeBehavior::step);
    }

    public void step() {
        if (Input.keyJustPressed(GLFW_KEY_ESCAPE)) {
            Game.stop();
        }
    }
}

File: Cookie/src/main/kotlin/extras/behaviors/QuitOnEscapeBehavior.kt
package extras.behaviors

import engine.core.Entity
import engine.core.Game
import engine.core.Input

import org.lwjgl.glfw.GLFW.GLFW_KEY_ESCAPE

class QuitOnEscapeBehavior : Entity() {

    fun step() {
        if (Input.keyJustPressed(GLFW_KEY_ESCAPE)) {
            Game.stop()
        }
    }

    companion object {

        init {
            Game.declareSystem(QuitOnEscapeBehavior::class.java) { obj -> obj.step() }
        }
    }
}
